src/Integration/FrameContextifierIntegration.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Sentry\Integration;
+
+use Psr\Log\LoggerInterface;
+use Sentry\Event;
+use Sentry\SentrySdk;
+use Sentry\Stacktrace;
+use Sentry\State\Scope;
+
+/**
+ * This integration reads excerpts of code around the line that originated an
+ * error.
+ *
+ * @author Stefano Arlandini <sarlandini@alice.it>
+ */
+final class FrameContextifierIntegration implements IntegrationInterface
+{
+    /**
+     * @var int The maximum number of lines of code to read
+     */
+    private $contextLines;
+
+    /**
+     * @var LoggerInterface|null A PSR-3 logger
+     */
+    private $logger;
+
+    /**
+     * Creates a new instance of this integration.
+     *
+     * @param int             $contextLines The maximum number of lines of code to read
+     * @param LoggerInterface $logger       A PSR-3 logger
+     */
+    public function __construct(int $contextLines = 5, ?LoggerInterface $logger = null)
+    {
+        if ($contextLines < 0) {
+            throw new \InvalidArgumentException(sprintf('The value of the $maxLinesToFetch argument must be greater than or equal to 0. Got: "%d".', $contextLines));
+        }
+
+        $this->contextLines = $contextLines;
+        $this->logger = $logger;
```

Comment:
Please use the `NullLogger` here to simplify usage below.

Reply:
I will, but it's becoming more and more common to instantiate a `NullLogger` in every place where we now accept the logger and, even though I'm pretty sure it doesn't hurts performances, I don't think we should continue doing this rather than checking for `null` to avoid one more object